services/gateway/gateway-service.mjs

```diff
@@ -499,26 +501,34 @@ class GatewayService {
    */
   async _teleportPlayerInternal(socket, msg, allowOverlap=false) {
     const res = await socket.moveLock.acquire('move', async () => {
+      const player = socket.playerData;
+
       // If the player doesn't move at all, just return true.
-      if (socket.playerData.mapCoord.equalsTo(msg.mapCoord) && socket.playerData.ghostMode === msg.ghostMode) {
+      if (player.mapCoord.equalsTo(msg.mapCoord) && player.ghostMode === msg.ghostMode) {
         return true;
       }
 
-      const ret = await this._enterCoord(msg.mapCoord);
+      // If the player wants his/her next status is in ghost mode, ignore occupation check and no need to enterCoord().
+      if (msg.ghostMode) {
+        // empty
```

Comment:
If I understand correctly, the logic currently in use is that the player will increase the cell's count whether it is in ghost mode. If you want to change this logic, then maybe you'll need to modify the disconnect logic as well?

Reply:
You are right. I will check all the places where redis is accessed.